libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/code_cache.hpp

```diff
@@ -92,22 +94,31 @@ class code_cache_base {
 
       std::filesystem::path _cache_file_path;
       int                   _cache_fd;
+      std::atomic<uint64_t> _executing_id{0}; // id of executing action
 
       io_context _ctx;
-      local::datagram_protocol::socket _compile_monitor_write_socket{_ctx};
+      local::datagram_protocol::socket _compile_monitor_write_socket{_ctx}; // protected by _mtx for async
```

Comment:
It is called from both the main application thread and the async thread.
For example:
main thread:
https://github.com/AntelopeIO/spring/blob/GH-1039-interrupt-block-validation/libraries/chain/webassembly/runtimes/eos-vm-oc/code_cache.cpp#L224-L224
async thread:
https://github.com/AntelopeIO/spring/blob/GH-1039-interrupt-block-validation/libraries/chain/webassembly/runtimes/eos-vm-oc/code_cache.cpp#L109-L109

Reply:
I guess in this case it's technically true because `native_handle()` isn't defined as being thread safe (though clearly in the implementations they are, as it's just an `int` or `HANDLE`). Otherwise afaict everything else seems safe. But it doesn't seem like it really changes much to just put it behind the mutex too